module/solutions/helper.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * name : helper.js
+ * author : Aman
+ * created-date : 03-sep-2020
+ * Description : Solution related helper functionality.
+ */
+
+/**
+    * SolutionsHelper
+    * @class
+*/
+module.exports = class SolutionsHelper {
+
+    /**
+   * Solution Data
+   * @method
+   * @name solutionDocuments
+   * @param {Array} [filterQuery = "all"] - solution ids.
+   * @param {Array} [fieldsArray = "all"] - projected fields.
+   * @param {Array} [skipFields = "none"] - field not to include
+   * @returns {Array} List of solutions. 
+   */
+  
+  static solutionDocuments(
+    filterQuery = "all", 
+    fieldsArray = "all",
+    skipFields = "none"
+  ) {
+    return new Promise(async (resolve, reject) => {
+        try {
+    
+            let queryObject = (filterQuery != "all") ? filterQuery : {};
+    
+            let projection = {}
+    
+            if (fieldsArray != "all") {
+                fieldsArray.forEach(field => {
+                    projection[field] = 1;
+                });
+            }
+
+            if( skipFields !== "none" ) {
+              skipFields.forEach(field=>{
+                projection[field] = 0;
+              })
+            }
+    
+            let solutionDocuments = 
+            await database.models.solutions.find(
+              queryObject, 
+              projection
+            ).lean();
+            
+            return resolve(solutionDocuments);
+            
+        } catch (error) {
+            return reject(error);
+        }
+    });
+  }
+
+  /**
+   * Create solution.
+   * @method create
+   * @name create
+   * @param {Object} data - solution creation data.
+   * @returns {JSON} solution creation data. 
+   */
+  
+  static create(data) {
+    return new Promise(async (resolve, reject) => {
+        try {
+    
+            let solutionData = 
+            await database.models.solutions.create(
```

Comment:
@aman-tunerlabs - Will the createdBy and updatedBy etc be set properly?

Reply:
added updatedBy. solution is not there throwing error.